components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationResponseProcessor.java

```diff
@@ -0,0 +1,244 @@
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.USER_ID_PATH:
+                authRequestUser.setId(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.EXTERNAL_ID_PATH:
+                authRequestUser.setExternalId(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.IDP_PATH:
+                authRequestUser.setIdp(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.SUB_PATH:
+                authRequestUser.setSub(castToString(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.USER_CLAIM_PATH:
+                authRequestUser.setUserClaims(castToClaim(performableOperation.getValue()));
+                break;
+            case AuthenticatorAdapterConstants.AuthRequestEntityPaths.USER_STORE_NAME_PATH:
+                userStoreDomain.setName(castToString(performableOperation.getValue()));
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Only the REPLACE operation is allowed. If any other operation is provided, it will be ignored
+     * and the authenticated user will not be updated.
+     *
+     * @param operation Operation from the external authentication service response.
+     */
+    private boolean isAllowedOperation(Operation operation) {
+
+        return Operation.REPLACE.equals(operation);
+    }
+
+    /**
+     * Create a new authenticated user based on the executed operations.
+     *
+     * @throws ActionExecutionResponseProcessorException    If error occurred while creating authenticated user
+     *                                                      due to invalid operation.
+     * @throws UserStoreException                           If error occurred while retrieving local user details.
+     */
+    public AuthenticatedUser createAuthenticateduser(AuthenticationRequestUser user, AuthenticationContext context,
+            UserStore userStore) throws ActionExecutionResponseProcessorException, UserStoreException {
+
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        authenticatedUser.setFederatedUser(user.resolveUserType());
+        authenticatedUser.setUserAttributes(user.resolveUserClaims());
+
+        if (authenticatedUser.isFederatedUser()) {
+            authenticatedUser.setFederatedIdPName(context.getExternalIdP().getIdPName());
+        } else {
+            /* User ID can only be set for local users. If the user is a federated user, will be ignored.
+             */
+            AbstractUserStoreManager userStoreManager = resolveUserStoreManager(context, userStore.getName());
+            authenticatedUser.setUserStoreDomain(userStore.getName());
```

Comment:
User store should ideally be optional.
If user store is not present and the user is still a local user we should be able to check if the user is present in the system (it goes over the chain of user stores)

Yet, this specific flow doesn't seem to function that way. It basically breaks the flow completely

Reply:
Fixed now.